### DISCUSSION TITLE
Using Google Play In-App Review API

### DISCUSSION THREAD

**Discussion by martinrademacher** [appcelerator/titanium_mobile#12715] [Q&A] [answered]
Hello,
is there a way to use the native Google Play In-App Review API with titanium?
There is a module for iOS (https://github.com/hansemannn/titanium-review-dialog) but it does not use the In-App review API on Android.
Thank you in advance for your help.

**Comment by m1ga** [comment, marked as answer]
are you sure?
https://github.com/hansemannn/titanium-review-dialog/blob/master/android/src/ti/reviewdialog/TitaniumReviewDialogModule.java#L50
should be the same as https://developer.android.com/guide/playcore/in-app-review/kotlin-java#launch-review-flow
not using it, just looked at the code 😄

**Reply by martinrademacher** [reply]
Thanks. I'll give it a try. I just looked at the documentation and I didn't see anything about the native android dialog.

**Reply by martinrademacher** [reply]
I finally found the time for that issue. iOS works as expected but compiling android terminates with the following error:
`[ERROR] [GRADLE] 
[ERROR] [GRADLE] FAILURE: Build failed with an exception.
[ERROR] [GRADLE] 
[ERROR] [GRADLE] * What went wrong:
[ERROR] [GRADLE] Execution failed for task ':app:checkDebugAarMetadata'.
[ERROR] [GRADLE] > Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
[ERROR] [GRADLE]    > Could not find com.github.hansemannn:smart-app-rate:1.1.1.`

**Reply by m1ga** [reply]
That might be a private repo or the version has changed. It was forked here: https://github.com/hansemannn/titanium-review-dialog/commit/34abd09c59bee2b70d21a1903428c82ea00c3d6a
I've changed it to the old dialog again: [ti.reviewdialog-android-2.0.1.zip](https://github.com/appcelerator/titanium_mobile/files/6553520/ti.reviewdialog-android-2.0.1.zip). You have to try that

**Reply by martinrademacher** [reply]
the error msg is gone but the review dialog on android does not show up. on iOS everything works fine.
This is the code I use to call the review dialog for android:
```javascript
                   Review.requestReview({
                        storeURL: 'https://play.google.com/store/apps/details?id=de.remind.fully',
                        onFeedback: event => {
                        // Send feedback to your server via "event.value"
                        }
                    });             
```
